packages/runtime/test/integration/pool-transaction-fee-test.ts

```diff
@@ -86,4 +97,291 @@ describe("Pooled Transaction Fees Test", function () {
+				toAccountAddr: bob.address,
+				amountMicroAlgos: amount,
+				payFlags: { totalFee: 0 }, // with 0 txn fee.
+			},
+			{
+				type: types.TransactionType.TransferAlgo,
+				sign: types.SignType.SecretKey,
+				fromAccount: bob.account,
+				toAccountAddr: john.address,
+				amountMicroAlgos: amount,
+				payFlags: { totalFee: 0 }, // with 0 txn fee.
+			},
+		];
+
+		runtime.executeTx(groupTx);
+
+		syncAccounts();
+		assert.equal(john.balance(), BigInt(initialBalance) - BigInt(fee));
+		assert.equal(alice.balance(), BigInt(initialBalance));
+		assert.equal(bob.balance(), BigInt(initialBalance));
+	});
+
+	it("Should unfunded accounts be able to issue transactions and opt-in", function () {
+		setupAsset();
+		const amount = 200000;
+		const fee = 3000;
+		// group with fee distribution
+		const groupTx: types.ExecParams[] = [
+			{
+				type: types.TransactionType.TransferAlgo,
+				sign: types.SignType.SecretKey,
+				fromAccount: john.account,
+				toAccountAddr: alice.address,
+				amountMicroAlgos: amount,
+				payFlags: { totalFee: 0 }, // with 0 txn fee
+			},
+			{
+				type: types.TransactionType.TransferAlgo,
+				sign: types.SignType.SecretKey,
+				fromAccount: alice.account,
+				toAccountAddr: elon.address,
+				amountMicroAlgos: amount,
+				payFlags: { totalFee: fee }, // this covers fee of entire group txns
+			},
+			{
+				type: types.TransactionType.OptInASA,
+				sign: types.SignType.SecretKey,
+				fromAccount: elon.account, // unfunded account
+				assetID: assetId,
+				payFlags: { totalFee: 0 }, // with 0 txn fee
+			},
+		];
+
+		runtime.executeTx(groupTx);
+
+		syncAccounts();
+		assert(elon.balance() !== BigInt(0));
+		assert.equal(john.balance(), BigInt(initialBalance) - BigInt(amount));
+		assert.equal(alice.balance(), BigInt(initialBalance) - BigInt(fee));
+		assert.equal(elon.balance(), BigInt(amount)); // unfunded account
+		// verify holding
+		assert.isDefined(elon.getAssetHolding(assetId));
+	});
+
+	it("Should not fail when account in first txn of group txn is unfunded account and trying to opt-in", function () {
+		setupAsset();
+		const amount = 200000;
+		const fee = 3000;
+		// group with fee distribution
+		const groupTx: types.ExecParams[] = [
+			{
+				type: types.TransactionType.OptInASA,
+				sign: types.SignType.SecretKey,
+				fromAccount: elon.account, // unfunded account
+				assetID: assetId,
+				payFlags: { totalFee: 0 }, // with 0 txn fee
+			},
+			{
+				type: types.TransactionType.TransferAlgo,
+				sign: types.SignType.SecretKey,
+				fromAccount: john.account,
+				toAccountAddr: alice.address,
+				amountMicroAlgos: amount,
+				payFlags: { totalFee: 0 }, // with 0 txn fee
+			},
+			{
+				type: types.TransactionType.TransferAlgo,
+				sign: types.SignType.SecretKey,
+				fromAccount: alice.account,
+				toAccountAddr: elon.address,
+				amountMicroAlgos: amount,
+				payFlags: { totalFee: fee }, // this covers fee of entire group txns
+			}
+		];
+
```

Comment:
lets add assert before that Elon balance is zero and that it indeed opted in afterwards

Reply:
Done